### DISCUSSION TITLE
Question: Future of qb with changes of qt's lts going closed source

### DISCUSSION THREAD

**Discussion by merspieler** [qutebrowser/qutebrowser#6021] [Q&A] [answered]
What are the consequences for qutebrowser of qt closing the sources of the 5.15 lts?

See https://www.theregister.com/2021/01/05/qt_lts_goes_commercial_only/

**Comment by The-Compiler** [comment, marked as answer]
I wouldn't be too worried about it. The main thing making this special is that the current Qt release is 6.0, but QtWebEngine isn't going to be part of Qt 6.x [until 6.1 or 6.2](https://www.qt.io/blog/add-on-support-in-qt-6.0-and-beyond).

This means we're stuck with the "status quo" of Qt 5.15 until [April or September](https://lists.qt-project.org/pipermail/development/2020-November/040704.html), most likely. This sounds bad, but it isn't really that bad - for qutebrowser, the most important part of the [announcement](https://lists.qt-project.org/pipermail/development/2021-January/040798.html) (which wasn't quoted in the above article) is:

> Exception being the Qt WebEngine [...], which have a 3rd party LGPL dependency.

This means we'll continue to get updates for the web engine - mainly security backports, but also Chromium upgrades. Usually the same Chromium version is kept during a given patch release, but Qt 5.15 is an exception: Since Qt 6.x doesn't have QtWebEngine yet, it does get Chromium upgrades (and already did, Qt 5.15.2 upgraded from 80 to 83).

Thus, this only affects bugfixes for the other parts of Qt qutebrowser is using - i.e. mostly Qt Core, Widgets, SQL, etc. Bugs in those affecting qutebrowser are rather rare, and usually there's a workaround - security bugs affecting qutebrowser are even rarer, as QtWebEngine is using Chromium's rather than Qt's code for parsing, rendering, network access, etc. etc.

This still kind of sucks for distributions who'd like to ship bugfixes for their Qt releases - then again, distributions like Ubuntu or Debian Stable ship the same Qt version (usually without any bugfix release updates!) for up to 5 years and this works for many projects (for qutebrowser, that on its own is a bit problematic, because an outdated web engine [without security support](https://www.debian.org/releases/buster/amd64/release-notes/ch-information.en.html#browser-security) is problematic).

Distributions like Archlinux are usually open to backporting patches before a Qt release if I ask them to - I suppose if I (or someone else) did the backporting work if something really critical turned up, nothing would change there. The main pain point is perhaps the Windows/macOS releases of qutebrowser, since I use Qt's binaries rather than compiling my own - but again, the chances of something outside of QtWebEngine needing an urgent update is rather slim.

tl;dr: We'll manage, it's less of a problem than you might think because QtWebEngine is exempt.

**Reply by merspieler** [reply]
thanks for the explanation.
